lib/selection.js

```diff
@@ -0,0 +1,202 @@
+/** A selection of pieces to download.
+ * @typedef {Object} SelectionItem
+ * @property {number} from
+ * @property {number} to
+ * @property {number} offset
+ * @property {number} priority
+ * @property {function} notify
+ * @property {boolean} isStreamSelection
+ */
+
+export class Selection {
+  /**
+   * @type {Array<SelectionItem>}
+   * @private
```

Comment:
skip these @private and @public jsdocs, we'd use # if we wanted to declare private, but we don't because we want people to access internal functions if they wish to, which is why _ is used for private, for strict engines such as TS, touching this will throw errors, which is annoying

Reply:
You're right, thanks for this. I removed them all